Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -847,13 +847,26 @@ class BrowserViewController: UIViewController {
     }
 
     func openURLInNewTab(url: NSURL) {
-        let tab: Browser
         if #available(iOS 9, *) {
-            tab = tabManager.addTab(NSURLRequest(URL: url), isPrivate: tabTrayController?.privateMode ?? false)
+            openURLInNewTab(url, isPrivate: tabTrayController?.privateMode ?? false)
         } else {
-            tab = tabManager.addTab(NSURLRequest(URL: url))
+            tabManager.addTabAndSelect(NSURLRequest(URL: url))
         }
-        tabManager.selectTab(tab)
+    }
+
+    @available(iOS 9, *)
+    func openURLInNewTab(url: NSURL?, isPrivate: Bool) {
+        let request: NSURLRequest?
+        if let url = url {
+            request = NSURLRequest(URL: url)
+        } else {
+            request = nil
+        }
+        let tabTrayController = self.tabTrayController ?? TabTrayController(tabManager: tabManager, profile: profile)
```

Comment:
Ugh - Can't wait to fix this when I invert the BVC. Storing the private state in the TVC is such a hack :(